flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/InternalWatermarkCallbackService.java

```diff
@@ -58,7 +59,17 @@
 	 * An array of sets of keys keeping the registered keys split
 	 * by the key-group they belong to. Each key-group has one set.
 	 */
-	private final Set<K>[] keysByKeygroup;
+	private final Set<K>[] registeredKeysByKeyGroup;
+
+	/**
+	 * An array of sets of keys keeping the keys "to delete" split
+	 * by the key-group they belong to. Each key-group has one set.
+	 * <p>
```

Comment:
IMHO, the formatting on paragraphs should be:
```
Some text bla bla black

<p>Other Text sdfasfaf
```

on the other hand, we don't have a style guide on that and no checkstyle for checking it .... 😉